src/Schedule.h

```diff
@@ -48,16 +48,34 @@ enum class TailStrategy {
      * case to handle the if statement. */
     GuardWithIf,
 
-    /** Guard the inner loop with an if statement that prevents
-     * evaluation beyond the original extent, with a hint that the
-     * if statement should be implemented with predicated operations.
-     * Always legal. The if statement is treated like a boundary
-     * condition, and factored out into a loop epilogue if possible.
-     * Pros: no redundant re-evaluation; does not constrain input our
+    /** Guard the loads and stores in the loop with an if statement
+     * that prevents evaluation beyond the original extent. Always
+     * legal. The if statement is treated like a boundary condition,
+     * and factored out into a loop epilogue if possible.
+     * Pros: no redundant re-evaluation; does not constrain input or
      * output sizes. Cons: increases code size due to separate
      * tail-case handling. */
     Predicate,
 
+    /** Guard the loads in the loop with an if statement that
+     * prevents evaluation beyond the original extent. Not legal
+     * for RVars, as it would change the meaning of the algorithm.
+     * The if statement is treated like a boundary condition, and
+     * factored out into a loop epilogue if possible.
+     * Pros: no redundant re-evaluation; does not constrain output
```

Comment:
I don't see how redundant re-evaluation is worse than adding extra computation on the end.

Also I think you mean does not constrain input sizes, not output